e2e_test/source/cdc/cdc.check_new_rows.slt

```diff
@@ -77,4 +77,39 @@ SELECT * FROM rw.products_test order by id limit 3
 query TTTT
 select order_date,customer_name,product_id,order_status from orders_no_backfill order by order_id;
 ----
-2022-12-01 15:08:22 Sam 110 0
\ No newline at end of file
+2022-12-01 15:08:22 Sam 110 0
+
+query II
+select * from numeric_to_varchar order by id;
+----
+1   3.14
+2   57896044618658097711785492504343953926634992332820282019728792003956564819967
+3   57896044618658097711785492504343953926634992332820282019728792003956564819968
+4   115792089237316195423570985008687907853269984665640564039457584007913129639936
+5   115792089237316195423570985008687907853269984665640564039457584007913129639936.555555
+6   NAN
+7   POSITIVE_INFINITY
+102 57896044618658097711785492504343953926634992332820282019728792003956564819967
+103 57896044618658097711785492504343953926634992332820282019728792003956564819968
+104 115792089237316195423570985008687907853269984665640564039457584007913129639936
+105 115792089237316195423570985008687907853269984665640564039457584007913129639936.555555
+106 NAN
+107 POSITIVE_INFINITY
```

Comment:
Supposed to be `NaN` and `Infinity`?

Reply:
🫠Too complicated to fix, as it requires us to know the types in upstream tables when parsing the data in JSON payload. Otherwise, we cannot distinguish whether a string is converted from a numeric or a real varchar. 

So in this PR, when mapping pg-numeric to varchar, NaN will be `NAN`, inf will be `POSITIVE_INFINITY `, and -inf will be `NEGATIVE_INFINITY `. At least these behavior is aligned among backfiling and normal data updates. Do you think that's acceptable?

For the bug in the conversion from pg-numeric to rw-numeric, I'll create a PR later to resolve it.

Reply:
Create a [issue](https://github.com/risingwavelabs/risingwave/issues/16395) here to record.